File: src/utils/leaderboard.js
const Discord = require("discord.js");

function createLeaderboardEmbed(data) {
  const draft = new Discord.MessageEmbed();
  const len = data.length;
  let rankingsDesc = "";

  if(len > 0) {
    draft.setTitle("Top 50 players based on points count");
    for(let i = 0; i < len; i++) {
      rankingsDesc += (i + 1).toString() + ". " + data[i].username + ": " + data[i].points;
      if(i < len - 1) {
        rankingsDesc += "\n";
      }
    }
    draft.setDescription(rankingsDesc);
  }
  else {
    draft.setDescription("Ranking list is empty. Go for the first!");
  }
  draft.setColor("#ff0000");

  return draft;
}

module.exports = {
  createLeaderboardEmbed
};

File: LeaderpointsBot.Client/Embeds/Leaderboard.cs
// Copyright (c) shigeru22, concept by Akshiro28.
// Licensed under the MIT license. See LICENSE in the repository root for details.

using System.Text;
using Discord;
using LeaderpointsBot.Database.Tables;
using LeaderpointsBot.Utils;

namespace LeaderpointsBot.Client.Embeds;

public static class Leaderboard
{
	public static Embed CreateLeaderboardEmbed(Users.UsersLeaderboardData[] data, DateTime lastUpdate, int usersLimit = 50, bool useLegacyColor = false)
	{
		string title = $"Top {usersLimit} players based on points count:";
		StringBuilder description = new StringBuilder();
		string footer = $"Last updated: {Date.DateTimeToString(lastUpdate, true)}";

		int dataLength = data.Length;
		for (int i = 0; i < dataLength; i++)
		{
			_ = description.Append($"{i + 1}. {data[i].Username}: {data[i].Points}{(i < dataLength - 1 ? "\n" : string.Empty)}");
			if (i == usersLimit - 1)
			{
				break;
			}
		}

		return new EmbedBuilder().WithTitle(title)
			.WithDescription(description.ToString())
			.WithFooter(footer)
			.WithColor(useLegacyColor ? LegacyBorderColor.Normal : BorderColor.Normal)
			.Build();
	}
}
